consul-lambda/consul-lambda-registrator/upsert_event.go

```diff
@@ -4,25 +4,63 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"strconv"
 
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/terraform-aws-consul-lambda/consul-lambda/structs"
 )
 
-const managedLambdaTag = "managed-by-lambda-registrator"
+const (
+	managedLambdaTag = "managed-by-lambda-registrator"
 
+	arnField                = "arn"
+	invocationModeField     = "invocationMode"
+	payloadPassthroughField = "payloadPassthrough"
+)
+
+// UpsertEvent struct holds data for an event that triggers the upserting of a Lambda function.
 type UpsertEvent struct {
 	structs.Service
+	EnvoyExtension
+}
+
+// EnvoyExtension configuration for an extension that patches Envoy resources for lambda
+type EnvoyExtension struct {
```

Comment:
suggest renaming this to `Arguments` or `LambdaArguments` because `EnvoyExtension` is the container, one level up, for this config. I was initially confused between this struct and `consul/api/EnvoyExtension`.

Reply:
that makes sense, I went with `LambdaArguments`